src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/IPackageSearchResultRenderer.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using NuGet.Configuration;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.CommandLine.XPlat
+{
+    /// <summary>
+    /// Defines methods for rendering the search results of NuGet packages.
+    /// </summary>
+    internal interface IPackageSearchResultRenderer
+    {
+        /// <summary>
+        /// Starts the rendering operation.
+        /// </summary>
+        void Start();
+
+        /// <summary>
+        /// Adds a list of packages from a source to be rendered.
+        /// </summary>
+        void Add(PackageSource source, IEnumerable<IPackageSearchMetadata> completedSearch);
```

Comment:
Since these actually output the result immediately, I wonder if something can indicate that with the naming?
```suggestion
        void Write(PackageSource source, IEnumerable<IPackageSearchMetadata> completedSearch);
```

Reply:
Write is an implementation detail, so it shouldn't be part of the contract. I think this is good.

Reply:
It's an implementation detail of the interface we're introducing. I get that it doesn't *have to* write, but since we *do write*, it seems confusing to me.

Reply:
I agree it's a little bit confusing now as there is one implementation of the class which immediately writes. However, once the `--format` option has been added, I plan to add other implementations of the renderer. For example, for json renderings, Add would not directly print the search results, rather it would add the search results to a json object.

Reply:
I see now that I understand your plans. 
It's probably ok as it is. Maybe a special word could be used for implementations that print?
Suggestion: `PackageSearchResultTablePrinter`